concrete/src/Authentication/Type/OAuth/GenericOauthTypeController.php

```diff
@@ -512,4 +519,28 @@ public function getUniqueId()
     abstract public function handle_attach_attempt();
 
     abstract public function handle_attach_callback();
+
+    public function handle_detach_attempt()
+    {
+        $user = new User();
+        if (!$user->isLoggedIn()) {
+            id(new RedirectResponse(\URL::to('')))->send();
+            exit;
+        }
+
+        $uID = $user->getUserID();
+        $namespace = $this->getHandle();
+        $binding = $this->getUniqueId();
+
+        $db = \Database::connection();
```

Comment:
It would be nice if this used a modern pattern.

Reply:
Fixed on https://github.com/concrete5/concrete5/pull/6018/commits/f162516be17421c2a0a5804868c63f839ec27f80